tests/ssl-opt.sh

```diff
@@ -358,7 +358,7 @@ requires_protocol_version() {
 
 # Space-separated list of ciphersuites supported by this build of
 # Mbed TLS.
-P_CIPHERSUITES=" $($P_CLI --help 2>/dev/null |
+P_CIPHERSUITES=" $($P_CLI help_ciphersuites 2>/dev/null |
```

Comment:
`ssl_client2 --help` returns a nonzero status, but this did not cause the script to abort, despite using `set -e`, because the nonzero status is on the left-hand side of a pipe and the status of the pipeline is just the status of the right-hand side. (The rationale for that design is to ignore SIGPIPE if it happens to the left-hand side.)

We should add some other sanity check here. We do expect at least one cipher suite:
```
if [ -z "${P_CIPHERSUITES# }" ]; then
    echo >&2 "$0: fatal error: no cipher suites found!"
    exit 125
fi
```

Reply:
Yes better to make that more robust. Actually I think with the new help we can get rid of at least one pipe, maybe both. I will try that.